src/cascade/dismod/db/metadata.py

```diff
@@ -0,0 +1,519 @@
+    child_nslist_id = Column(None, ForeignKey("nslist.nslist_id"), nullable=True)
+    """If not null, identifies a list of node and smoothing pairs. The
+    node_id for each of the children must appear in the list. The corresponding
+    smoothing is used for that child and the rate corresponding to this
+    row of the rate table."""
+
+
+class MulCovEnum(enum.Enum):
+    rate_value = 0
+    meas_value = 1
+    meas_std = 2
+
+
+class MulCov(Base):
+    """Covariate multipliers and their priors.
+
+    If this is empty, there are no covariate multipliers in the model.
+    """
+
+    __tablename__ = "mulcov"
+
+    mulcov_id = Column(Integer(), primary_key=True)
+    mulcov_type = Column(Enum(MulCovEnum), nullable=False)
+    rate_id = Column(None, ForeignKey("rate.rate_id"), nullable=True)
+    """Determines the rate that this covariate and multiplier affects.
+    If mulcov_type is of type meas_value or meas_std, this must be null."""
+    integrand_id = Column(None, ForeignKey("integrand.integrand_id"), nullable=True)
+    covariate_id = Column(None, ForeignKey("covariate.covariate_id"), nullable=False)
+    smooth_id = Column(None, ForeignKey("smooth.smooth_id"), nullable=True)
+    """If this is null, the covariabe multiplier is always zero and no
+    model_variables are allocated for it."""
+
+
+class AvgInt(Base):
+    __tablename__ = "avgint"
+
+    avgint_id = Column(Integer(), primary_key=True)
+    integrand_id = Column(None, ForeignKey("integrand.integrand_id"))
+    node_id = Column(None, ForeignKey("node.node_id"))
+    weight_id = Column(None, ForeignKey("weight.weight_id"))
+    age_lower = Column(Float())
+    age_upper = Column(Float())
+    time_lower = Column(Float())
+    time_upper = Column(Float())
+    # Greg label = Column(String())
+
+
+class Data(Base):
+    __tablename__ = "data"
+
+    data_id = Column(Integer(), primary_key=True)
+    data_name = Column(String(), unique=True)
+    """This is in the docs but not in the code."""
+
+    integrand_id = Column(None, ForeignKey("integrand.integrand_id"))
+    node_id = Column(None, ForeignKey("node.node_id"))
+    weight_id = Column(None, ForeignKey("weight.weight_id"))
+    age_lower = Column(Float())
+    age_upper = Column(Float())
+    time_lower = Column(Float())
+    time_upper = Column(Float())
+    hold_out = Column(Integer())
+    """Zero or one for hold outs during fit command"""
+    density_id = Column(Integer())
+    meas_value = Column(Float())
+    meas_std = Column(Float())
+    eta = Column(Float(), nullable=True)
+    nu = Column(Float(), nullable=True)
+    # Greg sample_size = Column(Float())
+
+
+class Option(Base):
+    __tablename__ = "option"
+
+    option_id = Column(Integer(), primary_key=True)
+    option_name = Column(String(), unique=True)
+    option_value = Column(String())
+
+
+class Constraint(Base):
+    """Greg shows this table, but it's not in Brad's docs."""
+
+    __tablename__ = "constraint_table"
```

Comment:
Should this just be called "constraint" or would that be a problem?

Reply:
Gah. Yes. I fixed one problem with these names which seems to have come out of issues with the documentation, but this is another that I didn't notice. We don't control these table names, they're Brad's. However this one (and the ones you point out below) are wrong. The `_table` suffix shouldn't be one there.

Reply:
ok, I stopped doing it after awhile, I am sure you noticed.